File: 백준/Bronze/11653. 소인수분해/소인수분해.py
N = int(input())
prime = []
num = N

i = 2
while i < N or num == 0:
    if num % i == 0:
        prime.append(i)
        num = num / i
    else:
        i += 1

if len(prime) == 0:
    if N != 1:
        prime.append(N)

for p in prime:
    print(p)

File: 백준/Bronze/11653. 소인수분해/소인수분해.java
import java.util.*;

public class Main{
   public static void main(String[] args)
   {
	   Scanner scanner = new Scanner(System.in);
	   
	   int num = scanner.nextInt();
	   while(num != 1)
	   {
		   for(int i = 2; i <= num; i++)
		   {
			   if(num%i == 0)
			   {
				   System.out.println(i);
				   num = num/i;
				   break;
			   }
		   }
	   }
	   
	   scanner.close();
   }
}